modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -141,14 +151,36 @@ public static void recoverDistributedSlotMap() throws AndesException {
      */
     private static void recoverMapsForEachQueue() throws AndesException {
         List<AndesQueue> queueList = contextStore.getAllQueuesStored();
-
-        for (AndesQueue queue : queueList) {
+        List<Future> futureSlotRecoveryExecutorList = new ArrayList<Future>();
+        ExecutorService executorService = Executors.newFixedThreadPool(5);
+        for (final AndesQueue queue : queueList) {
+            final String queueName = queue.queueName;
             // Skip slot creation for Dead letter Channel
-            if (DLCQueueUtils.isDeadLetterQueue(queue.queueName)) {
+            if (DLCQueueUtils.isDeadLetterQueue(queueName)) {
                 continue;
             }
-
-            initializeSlotMapForQueue(queue.queueName);
+            Future submit = executorService.submit(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        log.info("Slot restoring start in " + queueName);
+                        initializeSlotMapForQueue(queueName);
+                        log.info("Slot restoring end in " + queueName);
+                    } catch (AndesException e) {
+                        log.error("Error occurred in slot recovery.", e);
+                    }
+                }
+            });
+            futureSlotRecoveryExecutorList.add(submit);
+        }
+        for (Future slotRecoveryExecutor : futureSlotRecoveryExecutorList) {
+            try {
+                slotRecoveryExecutor.get();
+            } catch (InterruptedException e) {
+                log.error("Error occurred in slot recovery.", e);
```

Comment:
I think it would be best to propagate this RuntimeException. This can be done by calling
```
Thread.currentThread().interrupt();
```
Detailed explanation can be found at http://www.ibm.com/developerworks/library/j-jtp05236/#2.1